### DISCUSSION TITLE
Text from random row in the db

### DISCUSSION THREAD

**Discussion by fgionghi** [Budibase/budibase#3288] [Help]
Hi, just start to try budibase today and I'm really impressed. The only thing is that the tutorial in the docs is very simple compared to what it can be done.

However I just want to print a random row from the db as text.

In this moment the only way I have been able to print a row from the db (excluding a table) is:
- Data provider (Data: myDb)
- Repeater (and as provider I select the previously created provider)
- Headline with text: `{{ Repeater.Mydb.myColumn }}

In this way I have X headline with a row from the selected column. 

Questions:
- Why only the repeater has the field "Provider"? Am I wrong or I always need  a Dataprovider+repeater to display values from the db?
- I would like something like: Data Provider (and I select `myDb`) > headline ( and as text `DataProvider.myColumn`). 
- With the repeater or without it, how can I only dispaly one row and choose it randomly?

Hope to have explained myself, thank you.

**Comment by mjashanks** [comment]
Hey @fgionghi. Welcome!

I understand what you are saying about the Repeater. It is a pain, and its something that we plan to address. Really, we should only have a `Data Provider` + `Headline`, as you say.

Right now though, a repeater is necessary. To only display one row
1. Add a data provider with `Limit` set to `1`
2. Add a repeater (this will "Repeat" on your single row)
3. Add a headline.

We are going to look into adding a `☑️ Repeater` setting to the dataprovider which will allow this atuomatically (or something similar)

**Reply by fgionghi** [reply]
Awesome thank you. And do you have suggestion about the `random` thing?
I thought something like `{{ random 0 DataGrat.Rows Length }}` but DataGrat is not the db, but only the first line as you said.

**Reply by mjashanks** [reply]
hmm - I don't think there's going to be a reliable way of pulling a truly random record unless you're using a custom SQL query.

You'll want to 
1. `Limit = 1`
2. `Sort by some Randomish column` - you could use `_id`, if using the Budibase DB. This _might_ change as you add more records. But will pull back the same record every time if no records are added or removed.

**Reply by mjashanks** [reply]
If you really want to do this, you could
1. Create a new Text column
2. `Sort by` this column (as my comment above)
3. Create a new Automation to
- Run every few minutes with a CRON trigger
- Query a row `Sort by your column, limit 1`
- Generate a random bit of text, using a javascript action
- Update that same row's column to be your value generated in javascript

I.e. the automation is randomising the current top record, based on the new column.

Very hacky - but it might just work :)

**Reply by fgionghi** [reply]
Thank you, I was trying another approach, do you think it make sense?

First, I add a new column called `randomValue` on mydb and the value is `{{ random 0 100 }}`, and this is working. 
Then I already have a button and now its job is only save what I put in a textbox, but could I add a new action that update all the value of my `randomValue` column? 
And the new value would be `{{ random 0 100 }}`. 
Then I will sort my DataProvider for ascending. If this is not truly random it's not a problem. 
I have doubts on the "Update all rows to a random value" part. 
I will try tomorrow

**Reply by mjashanks** [reply]
Yea - there is no `Update All` at the moment. There is only `Update One`. Apart from that, your idea is better than mine :)
